service/history/ndc/history_replicator_test.go

```diff
@@ -1860,3 +1861,379 @@ func Test_applyNonStartEventsToNoneCurrentBranchWithoutContinueAsNew(t *testing.
+			mockTaskAffordance: func(mockTask *MockreplicationTask) {
+				mockTask.EXPECT().getLogger().Return(log.NewNoop()).Times(2)
+				mockTask.EXPECT().getDomainID().Return("test-domain-id").Times(1)
+				mockTask.EXPECT().getExecution().Return(&types.WorkflowExecution{
+					WorkflowID: "test-workflow-id",
+					RunID:      "test-run-id",
+				}).Times(1)
+				mockTask.EXPECT().getEvents().Return(nil).Times(1)
+				mockTask.EXPECT().getNewEvents().Return(nil).Times(1)
+			},
+			mockStateBuilderAffordance: func(mockStateBuilder *execution.MockStateBuilder) {
+				mockStateBuilder.EXPECT().ApplyEvents(
+					"test-domain-id",
+					gomock.Any(),
+					types.WorkflowExecution{
+						WorkflowID: "test-workflow-id",
+						RunID:      "test-run-id",
+					},
+					nil,
+					nil,
+				).Return(nil, fmt.Errorf("applyEvents error")).Times(1)
+			},
+			mockTransactionManagerAffordance: func(mockTransactionManager *MocktransactionManager) {},
+			mockShardContextAffordance:       func(mockShard *shard.MockContext) {},
+			expectError:                      fmt.Errorf("applyEvents error"),
+		},
+		"Case3: error during createWorkflow": {
+			mockTaskAffordance: func(mockTask *MockreplicationTask) {
+				mockTask.EXPECT().getLogger().Return(log.NewNoop()).Times(2)
+				mockTask.EXPECT().getDomainID().Return("test-domain-id").Times(1)
+				mockTask.EXPECT().getExecution().Return(&types.WorkflowExecution{
+					WorkflowID: "test-workflow-id",
+					RunID:      "test-run-id",
+				}).Times(1)
+				mockTask.EXPECT().getEvents().Return(nil).Times(1)
+				mockTask.EXPECT().getNewEvents().Return(nil).Times(1)
+				mockTask.EXPECT().getEventTime().Return(time.Now()).Times(1)
+			},
+			mockStateBuilderAffordance: func(mockStateBuilder *execution.MockStateBuilder) {
+				mockStateBuilder.EXPECT().ApplyEvents(
+					"test-domain-id",
+					gomock.Any(),
+					types.WorkflowExecution{
+						WorkflowID: "test-workflow-id",
+						RunID:      "test-run-id",
+					},
+					nil,
+					nil,
+				).Return(nil, nil).Times(1)
+			},
+			mockTransactionManagerAffordance: func(mockTransactionManager *MocktransactionManager) {
+				mockTransactionManager.EXPECT().createWorkflow(
+					gomock.Any(),
+					gomock.Any(),
+					gomock.Any(),
+				).Return(fmt.Errorf("createWorkflow error")).Times(1)
+			},
+			mockShardContextAffordance: func(mockShard *shard.MockContext) {},
+			expectError:                fmt.Errorf("createWorkflow error"),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			// Mock objects
+			mockTask := NewMockreplicationTask(ctrl)
+			mockExecutionContext := execution.NewMockContext(ctrl)
+			mockMutableState := execution.NewMockMutableState(ctrl)
+			mockStateBuilder := execution.NewMockStateBuilder(ctrl)
+			mockTransactionManager := NewMocktransactionManager(ctrl)
+			mockShard := shard.NewMockContext(ctrl)
+			logger := log.NewNoop()
```

Comment:
nit: I prefer testlogger.New(t) which can give you logs

Reply:
There are too many test cases. Switching to testLogger might cause the tests too messy :(